### PR TITLE
Bookings extensions conflicts show a more detailed error message

### PR DESCRIPTION
**Pull request by bryangaledxw**
# Changes in this PR

This PR adds improved error messages anywhere we handle 409 conflict errors. We do this by inspecting the error message from the API, extracting the ID of the conflicting booking/lost bed, and using that to construct an error that links to the thing causing the conflict

### Before
localhost_3000_properties_4f5ca40e-7ecb-4395-b1f9-0f26607a342c_bedspaces_737936b3-f305-4cbf-a79b-fba7ffcf08bf_bookings_new (1)

### After
localhost_3000_properties_4f5ca40e-7ecb-4395-b1f9-0f26607a342c_bedspaces_737936b3-f305-4cbf-a79b-fba7ffcf08bf_bookings_new

# Release checklist

[Release process
documentation](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/edit-v2/4247847062?draftShareId=a1c360ab-bd31-4db1-aae3-7cc002761de9).

## Pre-merge checklist

- [ ] Are any changes required to dependent services for this change to work?
  (eg. CAS API)
- [ ] Have they been released to production already?

## Post-merge checklist

- [ ] [Manually
  approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to test
- [ ] [Manually
  approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to preprod
- [ ] [Manually
  approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to prod

### CODE REVIEW

**Comment by bryangaledxw** [comment]
[Bookings extensions conflicts show a more detailed error message](https://trello.com/c/BiIJafWt/1023-bookings-extensions-conflicts-show-a-more-detailed-error-message)

File: server/utils/bookingUtils.test.ts [hunk 1]
```diff
@@ -391,4 +395,69 @@ describe('bookingUtils', () => {
...
+              premisesId,
+              roomId,
+              lostBedId,
+            })}">existing void</a>`,
+          },
+        ],
+      })
+    })
+
+    it('generates a bespoke for a single date', () => {
```
Comment (patrickjfl) [inline server/utils/bookingUtils.test.ts]: generates a bespoke error?

File: cypress_shared/pages/page.ts [hunk 2]
```diff
@@ -170,7 +174,7 @@ export default abstract class Page extends Component {
   }
 
   clickBreadCrumbUp(): void {
-    cy.get('li.govuk-breadcrumbs__list-item:nth-last-child(2)').click()
+    Page.clickBreadCrumbUp()
```
Comment (patrickjfl) [inline cypress_shared/pages/page.ts:177]: Is there an advantage to having this method call a Page method, rather than doing the operation directly?

**Review by patrickjfl — APPROVED** [review]
LGTM 👍

File: cypress_shared/pages/page.ts (hunk 2 above)
Comment (bryangaledxw) [reply cypress_shared/pages/page.ts:177]: Just to save a little bit of repetition basically

File: server/utils/bookingUtils.test.ts (hunk 1 above)
Comment (bryangaledxw) [reply server/utils/bookingUtils.test.ts]: Thanks!

**Pull request merged**